libpod/networking_linux.go

```diff
@@ -63,18 +63,65 @@ const (
 // This is need because a HostIP of 127.0.0.1 would now allow the gvproxy forwarder to reach to open ports.
 // For machine the HostIP must only be used by gvproxy and never in the VM.
 func (c *Container) convertPortMappings() []types.PortMapping {
-	if !machine.IsGvProxyBased() || len(c.config.PortMappings) == 0 {
+	if len(c.config.PortMappings) == 0 {
 		return c.config.PortMappings
 	}
-	// if we run in a machine VM we have to ignore the host IP part
+
+	var convert func(types.PortMapping) []types.PortMapping
+	if machine.IsGvProxyBased() {
+		// if we run in a qemu VM we have to ignore the host IP part
+		convert = stripHost
+	} else if machine.MachineHostType() == machine.Wsl {
+		// WSL machines do not relay ipv4 traffic to dual-stack ports, simulate instead
+		convert = splitDualStack
+	} else {
+		return c.config.PortMappings
+	}
+
 	newPorts := make([]types.PortMapping, 0, len(c.config.PortMappings))
 	for _, port := range c.config.PortMappings {
-		port.HostIP = ""
-		newPorts = append(newPorts, port)
+		newPorts = append(newPorts, convert(port)...)
 	}
+
 	return newPorts
 }
 
+func stripHost(port types.PortMapping) []types.PortMapping {
+	port.HostIP = ""
+	return []types.PortMapping{port}
+}
+
+func splitDualStack(port types.PortMapping) []types.PortMapping {
+	ports := []types.PortMapping{port}
+	protocol := port.Protocol
+	if strings.HasSuffix(protocol, "4") || strings.HasSuffix(protocol, "6") {
+		return ports
+	}
```

Comment:
This will never be the case see the other comment

Reply:
I mainly wanted to prevent a situation where if it was ever exposed since it would break the workaround. Although perhaps if it ever was supported it would be instead be implemented on a separate field. I am happy to delete this since its not part of the contract.